src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_ProfileStack/Profile_PrivacyScreen.tsx

```diff
@@ -98,7 +99,9 @@ export const Profile_PrivacyScreen = () => {
               }}
               testID="privacyButton"
               onPress={async () => {
-                await Linking.openURL(kettleInfo.privacyDashboardUrl!);
+                const privacyDashboardUrl = await getPrivacyDashboardUrl();
+                privacyDashboardUrl &&
+                  (await Linking.openURL(privacyDashboardUrl));
```

Comment:
Not really sure that this is the best handling for the user. Earlier we didn't show the button if there was no `privacyDashboardUrl`. Now we show the button, but clicking it has no effect if no `privacyDashboardUrl`. 

Is it possible to make the handling more safe than before, but still having the url's available on the kettle info object?

Reply:
I had a talk with Kogenta and seems like this method always returns a URL so that is why is there but if for some reason it fails no action as you said, before this URL was inside KettleInfo, but we moved it out because the URL must be generated every time is requested because it expires in 15m so in order to show the button based on if there is a URL or not is fine but might be that when the user really presses the button the link expires, so that is why is now upon requests.